sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -226,6 +226,86 @@ class RocksDB(
     this
   }
 
+  /**
+   * Load from the start snapshot version and apply all the changelog records to reach the
+   * end version. Note that this will copy all the necessary files from DFS to local disk as needed,
+   * and possibly restart the native RocksDB instance.
+   *
+   * @param startVersion version of the snapshot to start with
+   * @param endVersion end version
+   * @param readOnly whether the RocksDB instance is read-only
+   * @return A RocksDB instance loaded with the state endVersion replayed from startVersion
+   */
+  def load(startVersion: Long, endVersion: Long, readOnly: Boolean): RocksDB = {
```

Comment:
Same, either method name or param name.

Reply:
Also, I don't feel like you'll need to think this for both read-only and read-write case. Though I don't remember we have read-only version of RocksDB class implementation.

Reply:
I will make it for read only.